gapis/api/vulkan/resources.go

```diff
@@ -529,17 +540,78 @@ func (t *ImageObject) ResourceData(ctx context.Context, s *api.State) (*api.Reso
 			return api.NewResourceData(api.NewTexture(&api.Cubemap{Levels: cubeMapLevels})), nil
 		}
 
+		if t.Info.ArrayLayers > uint32(1) {
+			// 2D texture array
+			layers := make([]*api.Texture2D, int(t.Info.ArrayLayers))
+			for layerIndex := range layers {
+				imageLayer := t.Layers.Get(uint32(layerIndex))
+				levels := make([]*image.Info, len(imageLayer.Levels))
+				for levelIndex := range levels {
+					imageLevel := imageLayer.Levels.Get(uint32(levelIndex))
+					img := imageLevel.imageInfo(ctx, s, format)
+					if img == nil {
```

Comment:
In GLES we've made an effort to show partially populated textures. It's annoying to only see the image if all levels / layers are populated with data.

Reply:
4334638 changes all the early returns to `continue`.